src/ctr_logging.c

```diff
@@ -540,3 +564,110 @@ void sync_logs(void)
 		if (fsync(k8s_log_fd) < 0)
 			pwarn("Failed to sync log file before exit");
 }
+
+
+/* strlen("1997-03-25T13:20:42.999999999+01:00 stdout 9999999999 999999999 ") + 1 */
+#define TSSTREAMBUFLEN 128
+
+/*
+ * PROPOSED: CRI Stream Format, variable length file format
+ */
+static int set_k8s_stream_timestamp(char *buf, ssize_t bufsiz, ssize_t *tsbuflen, const char *pipename, uint64_t offset, ssize_t buflen,
+				    ssize_t *btbw)
+{
+	char off_sign = '+';
+	int off, len, err = -1;
+
+	struct timespec ts;
+	if (clock_gettime(CLOCK_REALTIME, &ts) < 0) {
+		/* If CLOCK_REALTIME is not supported, we set nano seconds to 0 */
+		if (errno == EINVAL) {
+			ts.tv_nsec = 0;
+		} else {
+			return err;
+		}
+	}
+
+	struct tm current_tm;
+	if (localtime_r(&ts.tv_sec, &current_tm) == NULL)
+		return err;
+
+	off = (int)current_tm.tm_gmtoff;
+	if (current_tm.tm_gmtoff < 0) {
+		off_sign = '-';
+		off = -off;
+	}
+
+	len = snprintf(buf, bufsiz, "%d-%02d-%02dT%02d:%02d:%02d.%09ld%c%02d:%02d %s %lud %ld ", current_tm.tm_year + 1900,
```

Comment:
(A drive-by comment with little context:) 

If the goal is to offload the CPU processing to consumers, shouldn’t this just be a timestamp instead of all the timezone lookups and formatting?

Either way, UTC would be better than ambiguous local time — does this one need custom parser code to get a `struct timespec` back?